lib/Controller/SettingsController.php

```diff
@@ -284,4 +305,169 @@ public function setPersonalSettings($templateFolder) {
+		}
+}
+
+	/**
+	 * @NoAdminRequired
+	 * @PublicPage
+	 * @NoCSRFRequired
+	 *
+	 * @param string $name
+	 * @return DataDisplayResponse
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	public function getFontFileOverview(string $name): DataDisplayResponse {
+		try {
+			$fontFileOverviewContent = $this->fontService->getFontFileOverview($name);
+			return new DataDisplayResponse(
+				$fontFileOverviewContent,
+				Http::STATUS_OK,
+				['Content-Type' => 'image/png']
+			);
+		} catch (NotFoundException $e) {
+			return new DataDisplayResponse('', Http::STATUS_NOT_FOUND);
+		}
+	}
+
+	/**
+	 * @param string $name
+	 * @return DataResponse
+	 * @throws NotFoundException
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	public function deleteFontFile(string $name): DataResponse {
+		$this->fontService->deleteFontFile($name);
+		return new DataResponse();
+	}
+
+	/**
+	 * @return JSONResponse
+	 */
+	public function uploadFontFile(): JSONResponse {
+		try {
+			$file = $this->getUploadedFile('fontfile');
+			if (isset($file['tmp_name'], $file['name'], $file['type'])) {
+				if (!in_array($file['type'], self::FONT_MIME_TYPES, true)) {
+					return new JSONResponse(['error' => 'Font type not supported'], Http::STATUS_BAD_REQUEST);
+				}
+				$newFileResource = fopen($file['tmp_name'], 'rb');
+				if ($newFileResource === false) {
+					throw new Exception('Could not read file');
+				}
+				$newFileName = $file['name'];
+				$uploadResult = $this->fontService->uploadFontFile($newFileName, $newFileResource);
+				return new JSONResponse($uploadResult);
+			}
+			return new JSONResponse(['error' => 'No uploaded file'], Http::STATUS_BAD_REQUEST);
+		} catch (Exception $e) {
+			$this->logger->error('Upload error', ['exception' => $e]);
+			return new JSONResponse(['error' => 'Upload error'], Http::STATUS_BAD_REQUEST);
+		}
+	}
+
+	/**
+	 * @param string $key
+	 * @return array
+	 * @throws UploadException
+	 */
+	private function getUploadedFile(string $key): array {
+		$file = $this->request->getUploadedFile($key);
+		$error = null;
+		$phpFileUploadErrors = [
+			UPLOAD_ERR_OK => $this->l10n->t('The file was uploaded'),
+			UPLOAD_ERR_INI_SIZE => $this->l10n->t('The uploaded file exceeds the upload_max_filesize directive in php.ini'),
+			UPLOAD_ERR_FORM_SIZE => $this->l10n->t('The uploaded file exceeds the MAX_FILE_SIZE directive that was specified in the HTML form'),
+			UPLOAD_ERR_PARTIAL => $this->l10n->t('The file was only partially uploaded'),
+			UPLOAD_ERR_NO_FILE => $this->l10n->t('No file was uploaded'),
+			UPLOAD_ERR_NO_TMP_DIR => $this->l10n->t('Missing a temporary folder'),
+			UPLOAD_ERR_CANT_WRITE => $this->l10n->t('Could not write file to disk'),
+			UPLOAD_ERR_EXTENSION => $this->l10n->t('A PHP extension stopped the file upload'),
+		];
+
+		if (empty($file)) {
+			$error = $this->l10n->t('No file uploaded or file size exceeds maximum of %s', [Util::humanFileSize(Util::uploadLimit())]);
```

Comment:
Can we throw the error directly instead of storing on `$error`? Then we won't need the last if-block.

Reply:
:+1: